### ISSUE TITLE
`std::mem::replace` suggestion deriving Clap `Subcommand`

### ISSUE THREAD

**Issue by hernangonzalez** [rust-lang/rust-clippy#11044]
### Summary

Hi,

I am getting this warning:
```
#[derive(Subcommand, Debug)]
  |          ^^^^^^^^^^
  |
  = note: or maybe you should use `std::mem::replace`?
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_swapped
note: the lint level is defined here
```

Can see the full file/project [here](https://github.com/hernangonzalez/codecrafters-git-rust/blob/284786123582fe6571f3f2e4411d01b50b83a8cf/src/cli.rs).

### Lint Name

almost_swapped

### Reproducer

I tried this code:

```rust
use crate::git::Sha;
pub use clap::Parser;
use clap::Subcommand;
use reqwest::Url;

#[derive(Subcommand, Debug)]
pub enum Command {
    Init,

    CatFile {
        #[arg(short = 'p')]
        part: String,
    },

    HashObject {
        #[arg(short = 'w')]
        file: String,
    },

    LsTree {
        #[arg(long = "name-only")]
        name_only: bool,

        sha: Sha,
    },

}
```

### Version

```text
rustc 1.70.0 (90c541806 2023-05-31)
binary: rustc
commit-hash: 90c541806f23a127002de5b4038be731ba1458ca
commit-date: 2023-05-31
host: x86_64-apple-darwin
release: 1.70.0
LLVM version: 16.0.2
```

**Comment by darklyspaced** [comment]
I attempted to reproduce the error that you are facing but I couldn't. I tried writing a MRE based on what you provided: 
```rust
pub use clap::Parser;
use clap::Subcommand;

#[warn(clippy::pedantic, clippy::correctness)]
#[derive(Subcommand, Debug)]
pub enum Command {
    Init,

    CatFile {
        #[arg(short = 'p')]
        part: String,
    },

    HashObject {
        #[arg(short = 'w')]
        file: String,
    },

    LsTree {
        #[arg(long = "name-only")]
        name_only: bool,
    },
}

fn main() {
    let mut a = 1;
    let mut b = 1;
    a = b;
    b = a;
}
```
But I didn't get manage to get clippy to complain about the `#[derive(Subcommand)].` It did however, correctly complain about the attempted swap of `a` and `b`.

I also cloned the latest commit of your repo and there seemed to be nothing wrong with it. Could you direct to me to a specific commit where I can see the problem?

Thanks!

**Comment by hernangonzalez** [comment]
Sadly enough, I do not see it anymore.. may have been a vscode glitch (?). Sorry for generating noise, I am closing this one now to avoid confusion. Will re-open if can trigger it again. Many thanks!!

**Comment by Centri3** [comment]
no, I believe this is actually still an issue. It still triggers in proc-macro generated code, it just shouldn't in `clap` due to the `allow`s. This should probably remain open, but with a different reproducer.
